agent/taskresource/firelens/firelens_unix.go

```diff
@@ -378,16 +461,62 @@ func (firelens *FirelensResource) generateConfigFile() error {
 		return errors.Wrap(err, "unable to generate firelens config")
 	}
 
+	confFilePath := filepath.Join(firelens.resourceDir, "config", "fluent.conf")
+	err = firelens.writeConfigFile(func(file oswrapper.File) error {
+		if firelens.firelensConfigType == FirelensConfigTypeFluentd {
+			return config.WriteFluentdConfig(file)
+		} else {
+			return config.WriteFluentBitConfig(file)
+		}
+	}, confFilePath)
+	if err != nil {
+		return errors.Wrapf(err, "unable to generate firelens config file")
+	}
+
+	seelog.Infof("Generated firelens config file at: %s", confFilePath)
+	return nil
+}
+
+// downloadConfigFromS3 downloads an external config file from S3 and saves it at ${RESOURCE_DIR}/config/external.conf.
+// The generated firelens config file fluent.conf will have a reference to include this file.
+func (firelens *FirelensResource) downloadConfigFromS3() error {
+	creds, ok := firelens.credentialsManager.GetTaskCredentials(firelens.executionCredentialsID)
+	if !ok {
+		return errors.New("unable to get execution role credentials")
+	}
+
+	bucket, key, err := s3.ParseS3ARN(firelens.externalConfigValue)
+	if err != nil {
+		return errors.Wrap(err, "unable to parse bucket and key from s3 arn")
+	}
+
+	s3Client, err := firelens.s3ClientCreator.NewS3ClientForBucket(bucket, firelens.region, creds.GetIAMRoleCredentials())
+	if err != nil {
+		return errors.Wrapf(err, "unable to initialize s3 client for bucket %s", bucket)
+	}
+
+	confFilePath := filepath.Join(firelens.resourceDir, "config", "external.conf")
+	err = firelens.writeConfigFile(func(file oswrapper.File) error {
+		return s3.DownloadFile(bucket, key, s3DownloadTimeout, file, s3Client)
+	}, confFilePath)
+
+	if err != nil {
+		return errors.Wrapf(err, "unable to download s3 config %s from bucket %s", key, bucket)
+	}
+
+	seelog.Infof("Downloaded firelens config file from s3 and saved to: %s", confFilePath)
```

Comment:
what's the value of this in info level?

Reply:
hmm can probably just be debug here.. will change